src/__init__.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+# from https://yann.lecun.com/exdb/mnist/
+def parse_mnist_data(
+        idx_file_training_samples: str,
+        idx_file_training_labels: str,
+        idx_file_test_samples: str,
+        idx_file_test_labels: str
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]: 
+    
+    training_samples = parse_mnist_images(idx_file_training_samples)
+    training_labels = parse_mnist_labels(idx_file_training_labels)
+
+    test_samples = parse_mnist_images(idx_file_test_samples)
+    test_labels = parse_mnist_labels(idx_file_test_labels)
+
+    return tuple[training_samples, training_labels, test_samples, test_labels]
+
+def parse_mnist_images(idx_file_path: str) -> np.ndarray:
+    f = open(idx_file_path, 'rb')
+
+    magic_num = int.from_bytes(f.read(2), 'big')
+    ftype = int.from_bytes(f.read(1), 'big')
+    num_dim = int.from_bytes(f.read(1), 'big')
+    num_img = int.from_bytes(f.read(4), 'big')
+    num_rows = int.from_bytes(f.read(4), 'big')
+    num_cols = int.from_bytes(f.read(4), 'big')
+
+    data = f.read()
+    out = np.ndarray((num_img, num_rows, num_cols), np.uint8, data)
+    print(out.shape)
+    return out
+
+def parse_mnist_labels(idx_file_path: str) -> np.ndarray:
+    f = open(idx_file_path, 'rb')
+    
+    magic_num = int.from_bytes(f.read(2), 'big')
+    ftype = int.from_bytes(f.read(1), 'big')
+    num_dim = int.from_bytes(f.read(1), 'big')
+    num_item = int.from_bytes(f.read(4), 'big')
+
+    data = f.read()
+    out = np.ndarray((num_item, 1), np.uint8, data)
+    print(out.shape)
+    return out
+
+def plot_image(img: np.ndarray) -> plt.figure:
+    assert len(img.shape) == 2, "input must be 2-dimensional (single image)"
+
+    fig, ax = plt.subplots()
+    ax.imshow(img)
+    plt.show()
```

Comment:
Hier kannst du auch mal bisschen mit dem styling von plots rumspielen. Wie schaffst du es zum Beispiel, dass
- die Achsenmarkierungen nicht mehr sichtbar sind
- die Farben im Graubereich liegen

Das ist das schöne an der Funktion, du musst es nur einmal definieren. :)
```suggestion
    fig, ax = plt.subplots()
    ax.imshow(img)
    plt.show()
```